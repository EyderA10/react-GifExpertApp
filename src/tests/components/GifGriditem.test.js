import React from 'react';
const { shallow } = require("enzyme")
const { default: GifGridItem } = require("../../components/GifGridItem")

describe('Pruebas con el componente GifGridItem', () => {

    const title = 'Game of thrones';
    const url = 'https://url.com';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('debe mostrar el componente correctamente', () => {


        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el title', () => {

        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);

    })

    test('debe de obtener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        //console.log(img.props().src);

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);


    })

    test('debe tener animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        // console.log(div.props());
        const prop = div.prop('className');
        expect(prop.includes('animate__fadeIn')).toBe(true);
    })
    
    
    
})
