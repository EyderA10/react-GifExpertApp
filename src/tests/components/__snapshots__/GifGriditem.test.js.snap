// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente GifGridItem debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Game of thrones"
    src="https://url.com"
  />
  <p
    className="animate__animated animate__bounce"
  >
    Game of thrones
  </p>
</div>
`;
